src/models/room-state.js

```diff
@@ -529,12 +531,12 @@ RoomState.prototype.getLastModifiedTime = function() {
 };
 
 /**
- * Get user IDs with the specified display name.
+ * Get user IDs with the specified or similar display names.
  * @param {string} displayName The display name to get user IDs from.
  * @return {string[]} An array of user IDs or an empty array.
  */
 RoomState.prototype.getUserIdsWithDisplayName = function(displayName) {
-    return this._displayNameToUserIds[displayName] || [];
+    return this._displayNameToUserIds[utils.removeHiddenChars(displayName)] || [];
```

Comment:
So its primarily because the API should handle this without you needing to, but having an internal one which avoids the duplication is probably the right way to go as the things being written internally call upon it: https://github.com/matrix-org/matrix-js-sdk/pull/668/files#diff-ff3076821c9d485ebf93690a38ed4692L737